images/macos/provision/core/action-archive-cache.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -e -o pipefail
+
```

Comment:
for macOS helpers are doubled, so you can use literally the same approach as I described for linux above :)